.circleci/config.yml

```diff
@@ -25,12 +33,129 @@ jobs:
       - run:
           name: Root package install
           command: npm ci
+      - save_cache:
+          key: deps-{{ checksum "package-lock.json" }}
+          paths:
+            - ./node_modules
+
+  lint:
+    working_directory: ~/stockflux
+    docker:
+      - image: circleci/node:10.15.1
+    steps:
+      - checkout:
+          path: ~/stockflux
+      - restore_cache:
+          key: deps-{{ checksum "package-lock.json" }}
       - run:
-          name: Bootstrap packages
+          name: Lint *.js files
+          command: npm run lint
+
+  build:
+    working_directory: ~/stockflux
+    docker:
+      - image: circleci/node:10.15.1
+    steps:
+      - checkout:
+          path: ~/stockflux
+      - restore_cache:
+          key: deps-{{ checksum "package-lock.json" }}
+      - run:
+          name: Restore Bitflux Cache
+          command: cd ~/stockflux/packages/stockflux-bitflux
+      - restore_cache:
+          key: deps-bitflux-{{ checksum "package-lock.json" }}
+      - run:
+          name: Restore Chart Cache
+          command: cd ~/stockflux/packages/stockflux-chart
+      - restore_cache:
+          key: deps-chart-{{ checksum "package-lock.json" }}
+      - run:
+          name: Restore Components Cache
+          command: cd ~/stockflux/packages/stockflux-components
+      - restore_cache:
+          key: deps-components-{{ checksum "package-lock.json" }}
+      - run:
+          name: Restore Container Cache
+          command: cd ~/stockflux/packages/stockflux-container
+      - restore_cache:
+          key: deps-container-{{ checksum "package-lock.json" }}
+      - run:
+          name: Restore Core Cache
+          command: cd ~/stockflux/packages/stockflux-core
+      - restore_cache:
+          key: deps-core-{{ checksum "package-lock.json" }}
+      - run:
+          name: Restore Launcher Cache
+          command: cd ~/stockflux/packages/stockflux-launcher
+      - restore_cache:
+          key: deps-launcher-{{ checksum "package-lock.json" }}
+      - run:
+          name: Restore News Cache
+          command: cd ~/stockflux/packages/stockflux-news
+      - restore_cache:
+          key: deps-news-{{ checksum "package-lock.json" }}
+      - run:
+          name: Bootstrap Packages
           command: npm run bootstrap
       - run:
-          name: Build packages
+          name: Build Packages
           command: npm run build
+      - run:
+          name: Save Bitflux Cache
+          command: cd ~/stockflux/packages/stockflux-bitflux
+      - save_cache:
+          key: deps-bitflux-{{ checksum "package-lock.json" }}
```

Comment:
A similar situation here, the `deps-bitflux-{{ checksum "package-lock.json" }}` key is using the root `~/stockflux/package-lock.json` file as the checksum in its cache key and we're actually caching the root `~/stockflux/node_modules` directory, which isn't what we want. I think the way this should be done would be to specify the full paths to the `~/stockflux/packages/stockflux-bitflux/package-lock.json` file for the checksum and `~/stockflux/packages/stockflux-bitflux/node_modules` directory for the `paths`. I think the build probably actually works because we're running `npm run bootstrap` above which installs all of the packages using Lerna.